app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java

```diff
@@ -262,59 +264,92 @@ private void deleteFile(String name, int position) {
-        final File fdelete = new File(name);
-        mFileList.remove(position);
-        notifyDataSetChanged();
-        StringUtils.getInstance().getSnackbarwithAction(mActivity, R.string.snackbar_file_deleted)
-                .setAction(R.string.snackbar_undoAction, v -> {
-                    if (mFileList.size() == 0) {
-                        mEmptyStateChangeListener.setEmptyStateInvisible();
-                    }
-                    updateDataset();
-                    undoClicked.set(1);
-                }).addCallback(new Snackbar.Callback() {
-                    @Override
-                    public void onDismissed(Snackbar snackbar, int event) {
-                        if (undoClicked.get() == 0) {
-                            fdelete.delete();
-                            mDatabaseHelper.insertRecord(fdelete.getAbsolutePath(),
-                                    mActivity.getString(R.string.deleted));
-                        }
-                    }
-                }).show();
-        if (mFileList.size() == 0)
-            mEmptyStateChangeListener.setEmptyStateVisible();
+        AlertDialog.Builder dialogAlert = new AlertDialog.Builder(mActivity)
+                .setCancelable(true)
+                .setNegativeButton(R.string.cancel, (dialogInterface, i) -> dialogInterface.dismiss())
+                .setTitle(R.string.delete_alert)
+                .setPositiveButton(R.string.yes, (dialog, which) -> {
+                    AtomicInteger undoClicked = new AtomicInteger();
+                    final File fdelete = new File(name);
+                    mFileList.remove(position);
+                    notifyDataSetChanged();
+                    StringUtils.getInstance().getSnackbarwithAction(mActivity, R.string.snackbar_file_deleted)
+                            .setAction(R.string.snackbar_undoAction, v -> {
+                                if (mFileList.size() == 0) {
+                                    mEmptyStateChangeListener.setEmptyStateInvisible();
+                                }
+                                updateDataset();
+                                undoClicked.set(1);
+                            }).addCallback(new Snackbar.Callback() {
+                                    @Override
+                                    public void onDismissed(Snackbar snackbar, int event) {
+                                        if (undoClicked.get() == 0) {
+                                            fdelete.delete();
+                                            mDatabaseHelper.insertRecord(fdelete.getAbsolutePath(),
+                                                    mActivity.getString(R.string.deleted));
+                                        }
+                                    }
+                            }).show();
+                    if (mFileList.size() == 0)
+                        mEmptyStateChangeListener.setEmptyStateVisible();
+                });
```

Comment:
There is code repition for deleting single file & multiple files.
Instead, can we have a common utility function for both with different values as it's parameters?


Also, alert heading https://github.com/Swati4star/Images-to-PDF/blob/978ad0f164aa5565bba5498ae52089b6b20be01f/app/src/main/res/values/strings.xml#L229 says "files" but it should say "file" when deleting single.

Reply:
I have updated the PR. Please check.